src/components/Products/Filters/Sizes.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import useHttpRequest from '../../../hooks/useHttpRequest';
+import Loading from '../../Main/Loading';
+
+const Sizes = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const [checked, setChecked] = useState([]);
+
+  const {
+    state: { error, data, loading }
+  } = useHttpRequest({
+    method: 'GET',
+    url: 'sizes/'
+  });
+
+  const query = () => {
+    const locationSearch = location.search;
+    const sizes = checked.join(',');
+
+    if (checked.length === 0 && locationSearch.includes('sizes')) {
+      if (locationSearch.includes('&sizes')) {
+        return navigate(locationSearch.replace(/&sizes=[^&]*/, ''));
+      }
+      return navigate(locationSearch.replace(/[?]sizes=[^&]*/, '?'));
+    }
+
+    if (
```

Comment:
Why are here so many checks? Simplify it. Something like:
if (checked.length) {
    navigate(locationSearch.includes('sizes') ? locationSearch.replace(/&sizes=[^&]*/, `&sizes=${sizes}`)) : locationSearch.concat(`$sizes=${sizes}`)
}

Reply:
Fixed. Thanks!